### ISSUE TITLE
Test On the Rinkeby

### ISSUE THREAD

**Issue by Skyge** [Skyge/ERC20-with-gas-token#2]
## Expect:

Write a test case for contracts of `transfer()`, `transferFrom()` and `approve()` and test on the Rinkeby network, easy enough, but need to finish it as soon as possible.

## Tool:

- Truffle
- [contracts](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/7ccd22456769c10da30632d7bbbdd732882d3dc8/contracts/token/ERC20/ERC20.sol)

*Notice: the version of the contract is `0.5.12`*

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 40.0 DAI (40.0 USD @ $1.0/DAI)  attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/Skyge/ERC20-with-gas-token/2/4481).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/Skyge/ERC20-with-gas-token/2/4481).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://chat.gitcoin.co/'>Gitcoin Chat</a>
* $166,818.92 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by touhonoob** [comment]
Isn't a test case already available here? https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/test/token/ERC20/ERC20.test.js

**Comment by Skyge** [comment]
I think it is a local test

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Workers have applied to start work__.


These users each claimed they can complete the work by 1 day, 7 hours from now.
Please review their action plans below:


**1) [surajsingla333](https://gitcoin.co/surajsingla333) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/Skyge/ERC20-with-gas-token/2/4481?mutate_worker_action=approve&worker=surajsingla333) | [reject worker](https://gitcoin.co/issue/Skyge/ERC20-with-gas-token/2/4481?mutate_worker_action=reject&worker=surajsingla333))_.**

I have worked on test driven development and will be able to do so within 2-3 hours. Will start the work rn.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/Skyge/ERC20-with-gas-token/2/4481).

**Comment by surajsingla333** [comment]
@Skyge I followed the previous issue and forked the chi repo as well and was working on the test cases. To perform tests on rinkeby network, you will have to fork the rinkeby network to ganache, I tried to do testing without forking the network but that didn't work and the only option is to fork the network. The test cases on the repo are suitable for testing the mentioned functions.

**Comment by Skyge** [comment]
@surajsingla333 You can use `infura` , `alchemy` or some tools else, then you can interact with the test net.

**Comment by surajsingla333** [comment]
> 
> 
> @surajsingla333 You can use `infura` , `alchemy` or some tools else, then you can interact with the test net.

Yes, I am using infura only. What I mean to day is, the tests which are already written can be tested on the rinkeby network only by forking the network. There is no other way to run the automated test in rinkeby network.

**Comment by Skyge** [comment]
@surajsingla333  So I can not run test case on the rinkeby directly, only by forking the network, right?

**Comment by surajsingla333** [comment]
> 
> 
> @surajsingla333 So I can not run test case on the rinkeby directly, only by forking the network, right?

Exactly!

**Comment by Skyge** [comment]
@surajsingla333 Without forking the network, have ran on the test net, such as
```
const ERC20 = artifacts.require("ERC20");
contract("DToken", (accounts) => {
    let token;
    before(async () => {
        token = await ERC20.at("0x123");
    }
    it('test : ', async function () {
        await token.mint(accounts[0], 5e18, {from: accounts[0]};
    }
})

```

**Comment by surajsingla333** [comment]
> accounts

@Skyge this worked fine in ganache network but in rinkeby or other testnets it was not working as expected because, it was not able to find accounts[0].

**Comment by Skyge** [comment]
@surajsingla333 You can set config in the `truffle-config.js` like this:
```
var HDWalletProvider = require("truffle-hdwallet-provider");

// Or, pass an array of private keys, and optionally use a certain subset of addresses
var privateKeys = [
  "3fxx80",
  "95xx90",
];

module.exports = {
    networks: {
        development: {
            host: "localhost",
            port: 7545,
            network_id: "*",
            gasPrice: 20000000000,
            gas: 8000000
        },
       rinkeby: {
            provider: () => new HDWalletProvider(privateKeys, `https://rinkeby.infura.io/v3/INFURA_KEY`, 0,2),
            network_id: 4, // ropsten's id
            gas: 6721975,
            gasPrice: 10000000000, // Gas price used for deploys: 10gwei
        }
    },
}
```

`truffle test --network rinkeby`

**Comment by surajsingla333** [comment]
> 
> 
> @surajsingla333 You can set config in the `truffle-config.js` like this:
> 
> ```
> var HDWalletProvider = require("truffle-hdwallet-provider");
> 
> // Or, pass an array of private keys, and optionally use a certain subset of addresses
> var privateKeys = [
>   "3fxx80",
>   "95xx90",
> ];
> 
> module.exports = {
>     networks: {
>         development: {
>             host: "localhost",
>             port: 7545,
>             network_id: "*",
>             gasPrice: 20000000000,
>             gas: 8000000
>         },
>        rinkeby: {
>             provider: () => new HDWalletProvider(privateKeys, `https://rinkeby.infura.io/v3/INFURA_KEY`, 0,2),
>             network_id: 4, // ropsten's id
>             gas: 6721975,
>             gasPrice: 10000000000, // Gas price used for deploys: 10gwei
>         }
>     },
> }
> ```
> 
> `truffle test --network rinkeby`

Yes @Skyge, I know how to configure the file. I replied to you only after performing all these steps. When we do `truffle test --network rinkeby` it throws error.

**Comment by Skyge** [comment]
@surajsingla333 what is the error, I do not show all the script file, but almost all.

**Comment by Skyge** [comment]
@surajsingla333 But as for me, it works, maybe you have a bad network connection.

**Comment by surajsingla333** [comment]
> 
> 
> @surajsingla333 But as for me, it works, maybe you have a bad network connection.

@Skyge can you send me the link to your test code?

**Comment by Skyge** [comment]
@surajsingla333  Sorry for replying late, I made a simple test at here: https://gist.github.com/Skyge/94d81ed240584259096792d77b50e7e5.

**Comment by surajsingla333** [comment]
> 
> 
> @surajsingla333 Sorry for replying late, I made a simple test at here: https://gist.github.com/Skyge/94d81ed240584259096792d77b50e7e5.

I am using the same configuration. I am using the [CHI](https://github.com/CryptoManiacsZone/chi) tokens (as mentioned in the previous issue) to test minting and other functions and they require the hight gas limit, I think that's why I am getting the error.

**Comment by Skyge** [comment]
`CHI` token is another issue, in this issue, just run test on the rinkeby with truffle, cause I expect someone can finish it in one day, so now it has expired, and would you mind stopping the work at the gitcoin?

**Comment by surajsingla333** [comment]
> 
> 
> @surajsingla333`CHI` token is another issue, in this issue, just run test on the rinkeby with truffle, cause I expect someone can finish it in one day, so now it has expired, and would you mind stopping the work at the gitcoin?

@Skyge the gist which you sent above worked perfectly fine in the rinkeby testnet. My mistake that I worked on  `CHI` token to test it. 
Would stop the work on gitcoin right away,

**Comment by Skyge** [comment]
@surajsingla333 Okay, I am not sure for the `CHI` token, you know, these days, the gas price is really high, so I am trying to reduce gas, so I think `CHI` token may be a good choice, I am not sure, maybe later I will also add a bounty for it, what do you think a gas token?

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Workers have applied to start work__.


These users each claimed they can complete the work by 3 days, 19 hours ago.
Please review their action plans below:


**1) [abhimanyu121](https://gitcoin.co/abhimanyu121) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/Skyge/ERC20-with-gas-token/2/4481?mutate_worker_action=approve&worker=abhimanyu121) | [reject worker](https://gitcoin.co/issue/Skyge/ERC20-with-gas-token/2/4481?mutate_worker_action=reject&worker=abhimanyu121))_.**

easy enough, wont take much time

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/Skyge/ERC20-with-gas-token/2/4481).

**Comment by surajsingla333** [comment]
> 
> 
> @surajsingla333 Okay, I am not sure for the `CHI` token, you know, these days, the gas price is really high, so I am trying to reduce gas, so I think `CHI` token may be a good choice, I am not sure, maybe later I will also add a bounty for it, what do you think a gas token?

@Skyge  When I was testing the `CHI` tokens. The default gas limit and gas price is set very high in their repo and if I try with the normal settings it shows `gas limit exceed` error. Will have to explore the same in details to see how the contract is working but currently, I can only say that it will consumemuch more gas.

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Cancelled** 

<hr>

__The funding of 40.0 DAI  (40.0 USD @ $1.0/DAI) attached to this issue has been **cancelled** by the bounty submitter__

 
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://chat.gitcoin.co/'>Gitcoin Chat</a>
* $246,564.50 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)
